common/src/main/java/com/alibaba/nacos/common/utils/CollectionUtils.java

```diff
@@ -302,6 +308,61 @@ public static <T> T getOnlyElement(Iterable<T> iterable) {
         throw new IllegalArgumentException(buildExceptionMessage(iterator, first));
     }
     
+    /**
+     * Returns a {@link Map} mapping each unique element in the given {@link Collection} to an {@link Integer}
+     * representing the number of occurrences of that element in the {@link Collection}.
+     *
+     * <p>Only those elements present in the collection will appear as keys in the map.
+     *
+     * @param coll the collection to get the cardinality map for, must not be null
+     * @return the populated cardinality map
+     */
+    public static Map getCardinalityMap(final Collection coll) {
+        Map count = new HashMap(coll.size());
+        for (Iterator it = coll.iterator(); it.hasNext(); ) {
+            Object obj = it.next();
+            Integer c = (Integer) (count.get(obj));
+            if (c == null) {
+                count.put(obj, INTEGER_ONE);
+            } else {
+                count.put(obj, c + 1);
+            }
+        }
+        return count;
+    }
+    
+    /**
+     * Returns <tt>true</tt> iff the given {@link Collection}s contain exactly the same elements with exactly the same
+     * cardinalities.
+     *
+     * <p>That is, iff the cardinality of <i>e</i> in <i>a</i> is equal to the cardinality of <i>e</i> in <i>b</i>, for
+     * each element <i>e</i> in <i>a</i> or <i>b</i>.
+     *
+     * @param a the first collection, must not be null
+     * @param b the second collection, must not be null
+     * @return <code>true</code> iff the collections contain the same elements with the same cardinalities.
+     */
+    public static boolean isEqualCollection(final Collection a, final Collection b) {
```

Comment:
If you want the collection not null, I think check it before you use it is better.